From c1d952fd202055e1b2634ed030649d75524d7d23 Mon Sep 17 00:00:00 2001
From: Dimitri John Ledkov <dimitri.ledkov@surgut.co.uk>
Date: Thu, 6 Nov 2025 07:24:00 +0000
Subject: [PATCH] systemd: Allow more systemd apps to log

On boot a few systemd apps are denied to connect to logging.

```
AVC avc:  denied  { sendto } for  pid=193 comm="systemd-hwdb" path="/run/systemd/journal/socket" scontext=system_u:system_r:systemd_hwdb_t:s0 tcontext=system_u:system_r:syslogd_t:s0 tclass=unix_dgram_socket permissive=1
AVC avc:  denied  { sendto } for  pid=212 comm="systemd-oomd" path="/run/systemd/journal/socket" scontext=system_u:system_r:systemd_oomd_t:s0 tcontext=system_u:system_r:syslogd_t:s0 tclass=unix_dgram_socket permissive=1
AVC avc:  denied  { sendto } for  pid=242 comm="systemd-nsresou" path="/run/systemd/journal/socket" scontext=system_u:system_r:systemd_nsresourced_t:s0 tcontext=system_u:system_r:syslogd_t:s0 tclass=unix_dgram_socket permissive=1
AVC avc:  denied  { sendto } for  pid=333 comm="systemd-ssh-iss" path="/run/systemd/journal/socket" scontext=system_u:system_r:systemd_ssh_issue_t:s0 tcontext=system_u:system_r:syslogd_t:s0 tclass=unix_dgram_socket permissive=1
```

Grant them access to syslog & journald.
---
 policy/modules/system/systemd.te | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/policy/modules/system/systemd.te b/policy/modules/system/systemd.te
index a37ddf696..83a65dc16 100644
--- a/policy/modules/system/systemd.te
+++ b/policy/modules/system/systemd.te
@@ -1156,6 +1156,8 @@ files_pid_filetrans(systemd_ssh_issue_t, systemd_ssh_issue_var_run_t, dir)
 kernel_dgram_send(systemd_ssh_issue_t)
 kernel_read_sysctl(systemd_ssh_issue_t)
 
+logging_send_syslog_msg(systemd_ssh_issue_t)
+
 dev_read_sysfs(systemd_ssh_issue_t)
 dev_read_vsock(systemd_ssh_issue_t);
 
@@ -1373,6 +1375,8 @@ dev_write_kmsg(systemd_hwdb_t)
 
 kernel_dgram_send(systemd_hwdb_t)
 
+logging_send_syslog_msg(systemd_hwdb_t)
+
 systemd_read_efivarfs(systemd_hwdb_t)
 
 ########################################
@@ -1985,6 +1989,8 @@ fs_list_cgroup_dirs(systemd_oomd_t)
 fs_setattr_cgroup_dirs(systemd_oomd_t)
 fs_write_cgroup_dirs(systemd_oomd_t)
 
+logging_send_syslog_msg(systemd_oomd_t)
+
 optional_policy(`
     dbus_acquire_svc_system_dbusd(systemd_oomd_t)
     dbus_read_pid_sock_files(systemd_oomd_t)
@@ -2010,6 +2016,8 @@ manage_dirs_pattern(systemd_nsresourced_t, systemd_nsresourced_runtime_t, system
 
 kernel_dgram_send(systemd_nsresourced_t)
 
+logging_send_syslog_msg(systemd_nsresourced_t)
+
 dev_read_sysfs(systemd_nsresourced_t)
 
 fs_cgroup_write_memory_pressure(systemd_nsresourced_t)
-- 
2.51.2

